### PR TITLE
refactor: move safe code to safe.d

### PR DESCRIPTION
**Pull request by WalterBright**
expression.d is getting too big. Move `checkUnsafeAccess()` and `isSafeCast()`

### CODE REVIEW

File: src/safe.d [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        return false;
+    DotVarExp dve = cast(DotVarExp)e;
+    if (VarDeclaration v = dve.var.isVarDeclaration())
+    {
+        if (sc.intypeof || !sc.func || !sc.func.isSafeBypassingInference())
+            return false;
+
+        auto ad = v.toParent2().isAggregateDeclaration();
+        if (!ad)
+            return false;
```
Comment (andralex) [inline src/safe.d]: not covered

**Comment by andralex** [comment]
OK

**Comment by andralex** [comment]
@MartinNowak Is the Travis failure transient?

**Comment by wilzbach** [comment]
> Is the Travis failure transient?

Hmm I rekicked the failing Travis build, it had this error:

```
==============================
Test failed: Invalid UTF-8 sequence (at index 3)
make[1]: *** [test_results/runnable/test8182.d.out] Error 1
make[1]: *** Waiting for unfinished jobs....
```

File: src/safe.d (hunk 1 above)
Comment (WalterBright) [inline src/safe.d]: Yeah, but this is a refactoring PR. That line was not covered before, either. Should not mix refactoring with semantic changes.

**Comment by WalterBright** [comment]
It's passing now, though I didn't change anything.

**Pull request merged**